src/SnowAgingSnicarMod.F90

```diff
@@ -0,0 +1,273 @@
+         TemperatureSnowLayerTop    = TemperatureSoilSnow(SnowLayerTop)
+         TemperatureSnowLayerBottom = (TemperatureSoilSnow(LoopInd+1)*ThicknessSnowSoilLayer(LoopInd) &
+                                      + TemperatureSoilSnow(LoopInd)*ThicknessSnowSoilLayer(LoopInd+1)) &
+                                      / (ThicknessSnowSoilLayer(LoopInd)+ThicknessSnowSoilLayer(LoopInd+1))
+      else
+         TemperatureSnowLayerTop    = (TemperatureSoilSnow(LoopInd-1)*ThicknessSnowSoilLayer(LoopInd) &
+                                      + TemperatureSoilSnow(LoopInd)*ThicknessSnowSoilLayer(LoopInd-1)) &
+                                      / (ThicknessSnowSoilLayer(LoopInd)+ThicknessSnowSoilLayer(LoopInd-1))
+         TemperatureSnowLayerBottom = (TemperatureSoilSnow(LoopInd+1)*ThicknessSnowSoilLayer(LoopInd) &
+                                      + TemperatureSoilSnow(LoopInd)*ThicknessSnowSoilLayer(LoopInd+1)) &
+                                      / (ThicknessSnowSoilLayer(LoopInd)+ThicknessSnowSoilLayer(LoopInd+1))
+      endif
+
+      TemperatureGradient(LoopInd) = abs((TemperatureSnowLayerTop - TemperatureSnowLayerBottom) / ThicknessSnowSoilLayer(LoopInd))
+
+      ! snow density
+      SnowDensity = (SnowLiqWater(LoopInd)+SnowIce(LoopInd)) / ThicknessSnowSoilLayer(LoopInd)
+
+      ! make sure snow density doesn't drop below 50 (see SnowDensityInd below)
+      SnowDensity = max(50.0,SnowDensity)
+
+      ! best-fit table indecies
+      TemperatureInd            = nint((TemperatureSoilSnow(LoopInd)-223.15) / 5) + 1
+      TemperatureGradientInd    = nint(TemperatureGradient(LoopInd) / 10) + 1
+      SnowDensityInd            = nint((SnowDensity-50) / 50) + 1
+
+      ! boundary check:
+      if (TemperatureInd < idx_T_min) then
+         TemperatureInd = idx_T_min
+      endif
+      if (TemperatureInd > idx_T_max) then
+         TemperatureInd = idx_T_max
+      endif
+      if (TemperatureGradientInd < idx_Tgrd_min) then
+         TemperatureGradientInd = idx_Tgrd_min
+      endif
+      if (TemperatureGradientInd > idx_Tgrd_max) then
+         TemperatureGradientInd = idx_Tgrd_max
+      endif
+      if (SnowDensityInd < idx_rhos_min) then
+         SnowDensityInd = idx_rhos_min
+      endif
+      if (SnowDensityInd > idx_rhos_max) then
+         SnowDensityInd = idx_rhos_max
+      endif
+
+      ! best-fit parameters
+      bst_tau   = snowage_tau(SnowDensityInd,TemperatureGradientInd,TemperatureInd)
+      bst_kappa = snowage_kappa(SnowDensityInd,TemperatureGradientInd,TemperatureInd)
+      bst_drdt0 = snowage_drdt0(SnowDensityInd,TemperatureGradientInd,TemperatureInd)
+
+      !LvK extra boundary check, to prevent when using old restart file with lower SnowRadiusMin than current run
+      if (SnowRadius(LoopInd) < SnowRadiusMin) then
+         SnowRadius(LoopInd) = SnowRadiusMin
+      endif
+
+      ! change in snow effective radius, using best-fit parameters
+      dr_fresh = SnowRadius(LoopInd)-SnowRadiusMin
```

Comment:
@tslin2 Here why use SnowRadiusMin instead of SnowRadiusFresh? dr_fresh should be the difference from fresh snow grain size not minimum?

Reply:
OK, I see that in CLM, it uses SnowRadiusMin